include/kodlab_mjbots_sdk/common_header.h

```diff
@@ -31,4 +33,84 @@ static void EnableCtrlC() {
   sigaction(SIGINT, &sigIntHandler, NULL);
   CTRL_C_DETECTED = false;
 }
+
+/**
+ * @brief Simple object wrapper that caches the latest value until invalidated
+ * @tparam T typename
+ */
+template<typename T>
+class ValidatedCache
+{
+
+private:
+  /**
+   * @brief Validation status of \c data_
+   */
+  bool valid_;
+
+  /**
+   * @brief Data being stored
+   */
+  T data_;
+
+public:
+  /**
+   * @brief Invalid data constructor
+   */
+  ValidatedCache() : valid_(false) {}
+
+  /**
+   * @brief Valid data constructor
+   */
+  explicit ValidatedCache(const T &data) : data_(data), valid_(true) {}
+
+  /**
+   * @brief Returns the data status
+   * @return \c true if data is valid, \c false otherwise
+   */
+  bool valid() const { return valid_; }
+
+  /**
+   * @brief Marks the class invalid so that future calls will change the cached
+   * value
+   */
+  void invalidate() { valid_ = false; }
+
+  /**
+   * @brief Sets the class data and marks it valid
+   * @param data valid data
+   */
+  void set(const T &data)
+  {
+    data_ = data;
+    valid_ = true;
+  }
+
+  /**
+   * @brief Accessor for data (<b>does not check validity</b>)
+   * @return data
+   */
+  T get()
+  {
+    if (!valid_)
+    {
+      std::cerr << "[WARN] Returning invalid data." << std::endl;
```

Comment:
This mode of operation for get() where we allow but warn against invalid data is a design choice we made. I think we are happy with this since otherwise we would need to do something with null pointers or std::optional, but bringing it up jic.

Reply:
Agreed, this should work for our purposes.  However, a more general class could optionally throw an error.